src/pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -600,9 +602,14 @@ def __resolve_ckpt_dir(self, trainer: "pl.Trainer") -> None:
             # if no loggers, use default_root_dir
             ckpt_path = os.path.join(trainer.default_root_dir, "checkpoints")
 
-        ckpt_path = trainer.strategy.broadcast(ckpt_path)
+        return ckpt_path
 
-        self.dirpath = ckpt_path
+    def _find_last_checkpoints(self, trainer: "pl.Trainer") -> List[str]:
+        # find all checkpoints in the folder
+        ckpt_path = self.__resolve_ckpt_dir(trainer)
+        if self._fs.exists(ckpt_path):
+            return [os.path.normpath(p) for p in self._fs.ls(ckpt_path, detail=False) if self.CHECKPOINT_NAME_LAST in p]
```

Comment:
Why do you normpath?

Reply:
Because otherwise tests fail on Windows, because `self._fs.ls` uses `posixpath.separator` on local filesystem regardless of platform